pkg/controller/volume/attachdetach/reconciler/reconciler.go

```diff
@@ -129,6 +130,19 @@ func (rc *reconciler) syncStates() {
 	}
 }
 
+func (rc *reconciler) isExclusiveVolume(volumeSpec *volume.Spec) bool {
+	if volumeSpec.PersistentVolume == nil {
+		return false
+	}
+	// check if this volume can be attached to multiple PODs/nodes, if yes, return false
+	for _, accessMode := range volumeSpec.PersistentVolume.Spec.AccessModes {
```

Comment:
Do I understand you correct that you confirm that using the access mode from the PV is the right thing to do here?

There is a check for PV==nil above the loop. In that case I consider the volume to not be exclusive as I trust in other k8s components to not try to mount non PV volumes to different nodes.